pandas/core/internals/blocks.py

```diff
@@ -1447,6 +1447,40 @@ def quantile(
         result = ensure_block_shape(result, ndim=2)
         return new_block_2d(result, placement=self._mgr_locs)
 
+    def round(self, decimals: int, using_cow: bool = False) -> Block:
+        """
+        Rounds the values.
+        If the block is not of an integer or float dtype, nothing happens.
+        This is consistent with DataFrame.round behavivor.
+        (Note: Series.round would raise)
+
+        Parameters
+        ----------
+        decimals: int,
+            Number of decimal places to round to.
+            Caller is responsible for validating this
+        using_cow: bool,
+            Whether Copy on Write is enabled right now
+        """
+        if not self.is_numeric or self.is_bool:
+            return self.copy(deep=not using_cow)
+        # If self.values is a numpy array of type int
+        # and decimals >= 0, this is a no-op and numpy
+        # returns the same array, so need to set refs
+        # https://github.com/numpy/numpy/blob/486878b37fc7439a3b2b87747f50db9b62fea8eb/numpy/core/src/multiarray/calculation.c#L625-L636
+        refs = None
+        if not self.is_extension and self.dtype.kind == "i" and decimals >= 0:
```

Comment:
I'd prefer checking:

```
result = self.values.round(..)
if self.values is result:
   refs = self.refs
```

This would not depend on an implementation detail of numpy